src/text.c

```diff
@@ -565,8 +566,10 @@ print_alternate_output_progress(axel_t *axel, char *progress, int width,
 			} else
 				progress[offset] = '#';
 		}
+		spacelen = axel->conn[i].lastbyte * width / total - offset - 1;
+		// spacelen will be "-1" sometimes, avoid this case
 		memset(progress + offset + 1, ' ',
-		       axel->conn[i].lastbyte * width / total - offset - 1);
+		       spacelen < 0 ? 0 : spacelen);
```

Comment:
Instead of using a variable, this could be:

        max(0, axel->conn[i].lastbyte * width / total - offset - 1)

Reply:
Thanks for reply.
Commit updated. :)